dll/win32/uxtheme/png.cpp

```diff
@@ -0,0 +1,44 @@
+#include "uxthemep.h"
+
+#ifdef MSSTYLES_ACCEPT_PNG_IMAGES
+#include <ole2.h>
+#include <objidl.h>
+#include <gdiplus.h>
+#include <gdipluscolor.h>
+
+BOOL MSSTYLES_TryLoadPng(HINSTANCE hTheme, LPCWSTR szFile, LPCWSTR type, OUT HBITMAP *pngBitmap)
+{
+    DWORD msgId = S_OK;
+
+    HRSRC pngRes = FindResourceW(hTheme, szFile, type);
+    DWORD pngSize = SizeofResource(hTheme, pngRes);
+
+    const void* pngData = LockResource(LoadResource(hTheme, pngRes));
+
+    HGLOBAL pngAlloc = GlobalAlloc(GMEM_MOVEABLE, pngSize);
+
```

Comment:
```suggestion
    if (!pngAlloc)
        return FALSE;
```
(then continue as usual)